infra.lvm_tools/shrink_lv/roles/shrink_lv/tasks/check_device.yaml

```diff
@@ -0,0 +1,20 @@
+- name: Get the mount point info
+  ansible.builtin.set_fact:
+    _mount_info: "{{ ansible_facts.mounts | selectattr('device', 'equalto', item.device) }}"
+
+- name: Assert that the mount point exists
+  ansible.builtin.assert:
+    that: (_mount_info | length) == 1
+    fail_msg: "Mount point {{ item.device }} does not exist"
+
+- name: Assert that the filesystem is supported
+  ansible.builtin.assert:
+    that: _mount_info[0].fstype in ['ext4']
+    fail_msg: "Unsupported filesystem '{{ _mount_info[0].fstype }}' on '{{ item.device }}'"
+
+- name: Assert that the filesystem has enough free space
+  ansible.builtin.assert:
+    that: _mount_info[0].block_size * _mount_info[0].block_used < (item.size | ansible.builtin.human_to_bytes)
+    fail_msg: >
+      Requested size {{ item.size }} is smaller than currently used
```

Comment:
suggestion: `item.size | ansible.builtin.human_to_bytes`. The point is to show values of the same unit, and here you might be referencing Gb to bytes.

Reply:
Are you referring to the error message? If yes, then this is the value provided by the user and therefore the one I think should be in the message.

Reply:
Ah I see. Sorry I got confused with `human_readable` with `human_to_bytes`. Nothing to see here.. move along 😄